src/test/java/spoon/test/variable/VariableTest.java

```diff
@@ -37,6 +44,64 @@
 
 public class VariableTest {
 
+    @Test
+    public void test() {
+        // contract: one can use setType without having a very hard generics type checking error
+
+        // to get this done, we use the refactoring below
+        CtClass c = Launcher.parseClass("class C{ void f() {Object i=null; }}");
+        List<CtLocalVariable> localVariables = c.getElements(new TypeFilter<>(CtLocalVariable.class));
+        System.out.println(localVariables);
+
+        // before: "CtLocalVariable<?> lv" could not typecheck, with a hard generics error
+        // now it works; we can use both "CtLocalVariable<?> lv" and "CtLocalVariable lv"
+        // it's source level backward compatible for Spoon itself
+        CtLocalVariable<?> lv = localVariables.get(0);
+        lv.setType(c.getFactory().Type().createReference(Object.class));
+
+        CtLocalVariable lv2 = localVariables.get(0);
+        lv2.setType(c.getFactory().Type().createReference(Object.class));
+
+    }
+
+    @Test
```

Comment:
As a note for @I-Al-Istannen `@modeltest` and Sniper printer don't work together. We need to set the printer before building the model.

Reply:
This also won't be a `@ModelTest` as it actually calls `run` instead of buildModel. I omitted support for doing more involved operations on the launcher (run processes it, prettyprints and optionally checks for compilation errors). I am not sure we want to support that, but we probably could allow a `sniper = true` setting if tests need that.

Reply:
Ack, removed the @Test annotation per the comment below.